### ISSUE TITLE
Add support for microdnf

### ISSUE THREAD

**Issue by atombrella** [hadolint/hadolint#948]
- [ ] This is a bug report
- [X] This is a feature request
- [ ] I searched existing issues before opening this one

### Expected behavior

There's an alternative to `dnf`, which is a light-weight version of dnf. Please add the same check for a missing `microdnf clean all` as the last statement.

Possibly, checking also for missing `--nodocs` for install would be good. `setopt=install_weak_deps=1` as used in #945

Hadolint does not catch any warnings for this Dockerfile:

```
FROM registry.access.redhat.com/ubi8/ubi-minimal:8.8

RUN microdnf install -y httpd
```

### Actual behavior

No warnings.

**Comment by ivanov17** [comment]
@atombrella I see that `microdnf` support is already implemented:

https://github.com/hadolint/hadolint/blob/b264eaa92ca2fe7df22a480a67b1c16e6e3e5f7c/src/Hadolint/Rule/DL3038.hs#L25

https://github.com/hadolint/hadolint/blob/b264eaa92ca2fe7df22a480a67b1c16e6e3e5f7c/src/Hadolint/Rule/DL3040.hs#L31

https://github.com/hadolint/hadolint/blob/b264eaa92ca2fe7df22a480a67b1c16e6e3e5f7c/src/Hadolint/Rule/DL3041.hs#L27

But support for `dnf5` command is not yet implemented. This command, which replaces `microdnf`, is used in Fedora 38 (and above) minimal containers. See the [Fedora Wiki](https://fedoraproject.org/wiki/Changes/MajorUpgradeOfMicrodnf) for details.

**Comment by angegar** [comment]
I do not know if microdnf is fully supported. Indeed, I receive this error message : **DL3040 warning: `dnf clean all` missing after dnf command** while my code already contains **microdnf clean all**
